src/Microsoft.VisualStudio.Composition.AppHost/CreateComposition.cs

```diff
@@ -171,7 +171,7 @@ public override bool Execute()
                 if (!string.IsNullOrEmpty(this.DgmlOutputPath))
                 {
                     configuration.CreateDgml().Save(this.DgmlOutputPath);
-                    this.writtenFiles.Add(this.DgmlOutputPath);
+                    this.writtenFiles.Add(this.DgmlOutputPath!);
```

Comment:
I didn't think you'd need ! here since you check string.IsNullOrEmpty on line 171

Reply:
And you'd be right too, when targeting `netcoreapp3.1`. But we also target `netstandard2.0` and `net472` which apparently lack annotations on `string.IsNullOrEmpty`, so the C# compiler doesn't recognize that method's behavior for null value tracing. So I have to add this character to suppress a warning for the older frameworks.